server/protocols/webadmin/webadmin-mailbox-deleted-message-vault/src/main/java/org/apache/james/webadmin/vault/routes/DeletedMessagesVaultRestoreTask.java

```diff
@@ -72,17 +79,19 @@ void incrementErrorRestoreCount() {
     private final User userToRestore;
     private final RestoreService vaultRestore;
     private final AdditionalInformation additionalInformation;
+    private final Query query;
 
-    DeletedMessagesVaultRestoreTask(User userToRestore, RestoreService vaultRestore) {
+    DeletedMessagesVaultRestoreTask(RestoreService vaultRestore, User userToRestore, Query query, QueryDTO queryDTO) {
```

Comment:
I need to pass QueryDTO to the aditional information object. Otherwise i have to reverse Query to QueryDTO in the Task

Reply:
I don't understand why you need it, could you elaborate a bit more?
The rule of thumb should be: never ask the same information twice because then you'll be in trouble to check that both are actually identical.

Reply:
the constructor looks a bit ugly, but I cant find a better way:
- Query is servered for searching.
- QueryDTO is for generating additional information. 

I could pass only QueryDTO to the task. But then Query will be translated from QueryDTO inside a task, not inside the route. if there is any error happens in that translating process, then error messages will not be returned directly to the client because it now is the task's exception. Client will receive a failed task rather than direct bad request response. That is the cause I pass both of them to Task's constructor.

Query and QueryDTO could be identical, but there is no way to translate a Query object to a QueryDTO object (query object contains full of lambda instances, not data). In an opposite way, I will end up with the issue I mention above.

Reply:
So, DTOs are for input/output. The simple fact that you put it into a business object is a problem.
Another point is : when parsing a QueryDTO you create Query object that is the business object that express the original QueryDTO. Why the hell can't we generate a QueryDTO from a Query ? I mean, the way to go is to transform Query into a QueryDTO (or another kind of DTO) at the edge of our system, that is, when we serialize to JSON.
If the code doesn't allow for that, it's probably broken somewhere.

Reply:
I have to idea to solve this issue, so I propose to not putting query into additional information at the moment, and creating a ticket for adding at later, is that okay?